pystac_client/stac_api_io.py

```diff
@@ -50,33 +50,23 @@ def __init__(
         # TODO - this should super() to parent class
         self.session = Session()
         self.session.headers.update(headers or {})
-        self.session.params.update(parameters or {})
+
+        if hasattr(self.session.params, "update"):
+            self.session.params.update(parameters or {})  # type: ignore
+        else:
+            logger.error("Parameters could not be set.")
 
         self._conformance = conformance
 
-    def read_text(
-        self,
-        source: Union[str, Link],
-        *args: Any,
-        parameters: Optional[Dict[str, Any]] = None,
-        **kwargs: Any,
-    ) -> str:
+    def read_text(self, source: pystac.link.HREF, *args: Any, **kwargs: Any) -> str:
         """Read text from the given URI.
 
         Overwrites the default method for reading text from a URL or file to allow
         :class:`urllib.request.Request` instances as input. This method also raises
         any :exc:`urllib.error.HTTPError` exceptions rather than catching
         them to allow us to handle different response status codes as needed.
         """
-        if isinstance(source, str):
-            href = source
-            if bool(urlparse(href).scheme):
-                return self.request(href, *args, parameters=parameters, **kwargs)
-            else:
-                with open(href) as f:
-                    href_contents = f.read()
-                return href_contents
-        elif isinstance(source, Link):
+        if isinstance(source, Link):
```

Comment:
Per the typing, `pystac.link.HREF` is either a `str` or a `PathLike`, so can't be a `Link`?

Reply:
`class Link(PathLike):`

Reply:
Whoa, did not realize that. Don't know if I like it, but not our problem here. Disregard, thanks.